go/stats/counters.go

```diff
@@ -187,13 +229,61 @@ func (mc *MultiCounters) Add(names []string, value int64) {
 	mc.Counters.Add(mapKey(names), value)
 }
 
-// Set sets the value of a named counter. len(names) must be equal to
-// len(Labels)
-func (mc *MultiCounters) Set(names []string, value int64) {
+// ResetCounter resets the value of a named counter back to 0. len(names)
+// must be equal to len(Labels)
+func (mc *MultiCounters) ResetCounter(names []string) {
 	if len(names) != len(mc.labels) {
 		panic("MultiCounters: wrong number of values in Set")
 	}
-	mc.Counters.Set(mapKey(names), value)
+
+	mc.Counters.ResetCounter(mapKey(names))
+}
+
+// MultiGauges is a MultiCounters implementation where the values can go up and down
+type MultiGauges struct {
+	Gauges
+	labels []string
+}
+
+// NewMultiGauges creates a new MultiGauges instance, and publishes it
+// if name is set.
+func NewMultiGauges(name string, help string, labels []string) *MultiGauges {
+	t := &MultiGauges{
+		Gauges: Gauges{Counters{
+			counts: make(map[string]*int64),
+			help:   help,
+		}},
+		labels: labels,
+	}
+	if name != "" {
+		publish(name, t)
+	}
+
+	publishPullMultiGauges(t, name)
```

Comment:
I think you should be able to do this by registering a new var hook using `stats.Register`, and have that function make these calls. This way, there will be no need to spray these additional publish calls through all the types.